crates/cxx-qt-lib/src/gui/qguiapplication.rs

```diff
@@ -221,4 +236,39 @@ impl QGuiApplication {
     pub fn desktop_file_name() -> QString {
         ffi::qguiapplication_desktop_file_name()
     }
+
+    /// Returns the current state of the modifier keys on the keyboard. The current state is updated
+    /// synchronously as the event queue is emptied of events that will spontaneously change the
+    /// keyboard state (QEvent::KeyPress and QEvent::KeyRelease events).
+    ///
+    /// It should be noted this may not reflect the actual keys held on the input device at the time
+    /// of calling but rather the modifiers as last reported in an event.
+    /// If no keys are being held Qt::NoModifier is returned.
+    pub fn keyboard_modifiers() -> KeyboardModifiers {
```

Comment:
Digging into the Qt sources reveals that these functions access a static variable directly to read the keyboard modifiers.

This is unfortunately not thread safe.
Please make this a member function, so we can at least ensure it's called in the same thread that created the QGuiApplication.

The same applies to mouse_buttons and query_keyboard_modifiers. (and [set_]desktop_file_name for that matter).

Related to #1205 and #1216